002_simulation_structure/002_simulation_structure.tex

```diff
@@ -1,23 +1,109 @@
+ 
+Within the tower, agent abilities are listed below:
+
+\begin{enumerate}
+    \item View Food on Platform: The agent cannot view the food reserves of the platform unless it is at the agent’s floor or the floor below the agent.
+    \item Food Consumption: Agent’s cannot preserve food but the option to ‘destroy’ food (eat food without receiving notable utility) is valid. Agents can only eat food from the platform when it is at the agent’s floor.
+    \item Time Details: Agents do not have explicit access to timed parameters (reshuffle period, day length, time of platform on each floor etc.), however, can use memory to calculate these.
+    \item Food Details: Agents do not know the food on the platform at the beginning of each day.
+
+    \item Health Details: Agents do have a sense of their current health level and utility they are receiving from food, as well as how long they can go without food before dying.
+    
+\end{enumerate}
+
+Among one-another, agent abilities are listed below:
+
+\begin{enumerate}
+    \item Agents should be able to converse with one another and share information.
+    \item Agents should be able to form agreements which must be followed.
+    \item Agents should be able to differentiate between any new agents that appear locally.
+    \item The communication must be limited to single agent interactions and single messages cannot travel past a defined, local distance. 
+\end{enumerate}
 
 %%%%%%%%
-%%%%%%%% Message Passing
+%%%%%%%%System Design
 %%%%%%%%
-\section{Message Passing}\label{message_passing}
-Need a nice diagram for this
+\section{System Design}\label{system_design}
+
+Based on our initial observations and constraints, possible implementations were explored early on in the design process. Our design process evolved from designing for the simple use case of allowing simple agents to eat food and communicate with primitive messages, to scaling this up to accommodate multiple agents on multiple floors.
+
+\subsection{Ticks \& Timing Details}
+
+Discrete time in the simulation is represented by ticks, the smallest unit of time. During each tick all agents can perform an action such as eat or send a message. This means that the amount of ticks per day restricts the communication between the agents. Moreover, given that messages can only be passed from one floor to one below or above on each tick, we can conclude that certain messages will never get to their receiver. The amount of ticks per day is set in the initialisation of the simulation and it is the product of the ticks per floor and the number of floors. This method was preferred to having a constant amount of ticks a day because that would cause problems with division with the number of floors (decimal numbers).
+
+\subsection{Agent Configuration}
+
+Agent structures are separated into two different aspects; a base agent and a custom agent – both of which are required to represent a single agent.
```

Comment:
```suggestion
Agent structures are separated into two different aspects - a base agent and a custom agent, both of which are required to represent a single agent.
```

Reply:
Also structure or struct? You use struct below, but here structure seems way more general than referring to a code struct